es-es/crystal-es.html.markdown

```diff
@@ -0,0 +1,542 @@
+s + = "Crystal" # => "hola, Crystal": cadena
+s.object_id # => 142528472: UInt64
+
+# Interpolación
+"sum = # {1 + 2}" # => "sum = 3": cadena
+
+# Cadenas multilínea
+"Esta es una cadena de
+varias líneas "
+
+# Otra notación para cadenas de caracteres
+# que permite insertar comillas
+% (hola "mundo") # => "hola \" mundo \ ""
+
+# Símbolos
+# Son inmutables y reutilizables, están representados internamente por
+# un Int32. A menudo se usan en lugar de cadenas
+# cuando la identidad es más importante que el contenido
+
+: symbol.class # => Símbolo
+
+oración =: pregunta? # :"¿pregunta?" : Símbolo
+
+oración ==: pregunta? # => verdadero: Bool
+oración ==: exclamación! # => falso: Bool
+oración == "pregunta?" # => falso: Bool
+
+# Tablas
+
+[1, 2, 3] .class # => Array (Int32)
+[1, "hola", 'x']. Clase # => Matriz (Int32 | String | Char)
+
+# Se debe especificar un tipo para tablas vacías
+[] # Error de sintaxis: para matrices vacías use '[] de ElementType'
+[] de Int32 # => []: matriz (Int32)
+Array (Int32) .new # => []: Array (Int32)
+
+# Las matrices se pueden indexar
+array = [1, 2, 3, 4, 5] # => [1, 2, 3, 4, 5]: matriz (Int32)
+array [0] # => 1: Int32
+array [10] # arroja la excepción IndexError
+array [-6] # arroja la excepción IndexError
+array [10]? # => nulo: (Int32 | Nulo)
+array [-6]? # => nulo: (Int32 | Nulo)# Desde el final
+matriz [-1] # => 5
+
+# Con un índice de inicio y tamaño
+matriz [2, 3] # => [3, 4, 5]
+
+# O con un intervalo
+matriz [1..3] # => [2, 3, 4]
+
+# La adición a una tabla se puede hacer con el operador <<
+matriz << 6 # => [1, 2, 3, 4, 5, 6]
+
+# Elimina la última entrada
+array.pop # => 6
+matriz # => [1, 2, 3, 4, 5]
+
+# Elimina la primera entrada
+array.shift # => 1
+matriz # => [2, 3, 4, 5]
+
+# Compruebe si hay un elemento presente en la matriz
+array.includes? 3 # => verdadero
+
+# Sintaxis especial para una serie de cadenas de caracteres o símbolos
+% w (uno dos tres) # => ["uno", "dos", "tres"]: matriz (cadena)
+% i (uno dos tres) # => [: uno ,: dos ,: tres]: Matriz (Símbolo)
+
+# Existe una sintaxis especial para matrices y otros tipos
+# siempre que definan un método .new y # <<
+set = Set {1, 2, 3} # => [1, 2, 3]
+set.class # => Set (Int32)
+
+# Lo que está arriba es equivalente a:
+set = Set (typeof (1, 2, 3)) nuevo
+establecer << 1
+conjunto << 2
+conjunto << 3# Tablas asociativas
+
+{1 => 2, 3 => 4} .clase # => Hash (Int32, Int32)
+{1 => 2, 'a' => 3} .class # => Hash (Int32 | Char, Int32)
+
+# Se debe especificar un tipo para matrices asociativas vacías
+{} # Error de sintaxis
+{} de Int32 => Int32 # {}
+Hash (Int32, Int32) .nuevo # {}
+
+# El acceso es a través de una clave
+hash = {"color" => "verde", "número" => 5}
+hash ["color"] # => "verde"
+hash ["no_such_key"] # => Falta la clave hash: "no_such_key" (KeyError)
+desmenuzar [ "no_such_key"]? # => nulo
+
+# Compruebe si hay una clave
+hash.has_key? "color" # => verdadero
+
+# Notación especial para teclas si son símbolos
```

Comment:
If we follow with other places translations,  "teclas" should be "claves" , I think.